internal/resources/resource_host.go

```diff
@@ -774,9 +823,41 @@ func resourceMetalHostDelete(d *schema.ResourceData, meta interface{}) (err erro
 	}
 
 	ctx = p.GetContext()
-	_, err = p.Client.HostsApi.Delete(ctx, d.Id())
 
-	return err
+	if _, err := p.Client.HostsApi.Delete(ctx, d.Id()); err != nil {
+		// nolint:wrapcheck // defer func is wrapping the error.
+		return err
+	}
+
+	// host deletes are asynchronous in Metal svc and we can not delete terraform's
+	// reference to the host until it has really gone from Metal svc. If we delete the
+	// reference too early, or in the presence of errors, we will never be able to retry
+	// the delete operation from Terraform (since it has no reference to the resource).
+	deleteStateConf := &resource.StateChangeConf{
+		Pending: []string{
+			hStateDeleting,
+		},
+		Target: []string{
+			hStateDeleted,
+		},
+		Refresh: func() (interface{}, string, error) {
+			host, _, err := p.Client.HostsApi.GetByID(ctx, d.Id())
+			if err != nil {
+				return nil, "", fmt.Errorf("get host %v", d.Id())
+			}
+
+			return host, string(host.State), nil
+		},
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      thirtySeconds,
+		MinTimeout: tenSeconds,
+	}
+
+	if _, err := deleteStateConf.WaitForStateContext(ctx); err != nil {
```

Comment:
Host acceptance needs to be updated:  remove the existing wait loop and  various values for the timer block.
 - https://github.com/HewlettPackard/hpegl-metal-terraform-resources/blob/a6a38cbf06045d3f4888c36a68e34ff8cc8850ed/internal/acceptance_test/resource_host_test.go#L113